### PR TITLE
Enable artifact publication to maven

### PR DESCRIPTION
**Pull request by ravivj-db**
**PR Checklist**

- [x] A description of the changes is added to the description of this PR.
- [ ] If there is a related issue, make sure it is linked to this PR.
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added or modified a feature, documentation in `docs` is updated

**Description of changes**

Add settings to configure the maven release process for `unitycatalog` project in SBT.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -282,4 +284,125 @@ val generate = taskKey[Unit]("generate code from APIs")
...
+  versionStr.split("\\.").toList match {
+    case majorStr :: minorStr :: patchStr :: _ =>
+      (majorStr, minorStr, patchStr)
+    case _ => throw new Exception(s"Could not parse version for $version.")
+  }
+}
+
+
+// Release settings for artifact that contains only Java source code
+lazy val javaOnlyReleaseSettings = releaseSettings ++ Seq(
```
Comment (tdas) [inline build.sbt:322]: Lets add all of these as a scala file under projects/

File: build.sbt [hunk 2]
```diff
@@ -282,4 +284,125 @@ val generate = taskKey[Unit]("generate code from APIs")
...
+val log4jVersion = "2.23.1"
+
+/*
+ ********************
+ * Release settings *
+ ********************
+ */
+import ReleaseTransformations._
+
+lazy val skipReleaseSettings = Seq(
```
Comment (tdas) [inline build.sbt:296]: add skipReleaseSettings to CLI
add javaOnlyReleaseSettings to server and client

Comment (ravivj-db) [reply build.sbt]: Done. What about apiDocs ?

File: build.sbt (hunk 1 above)
Comment (ravivj-db) [reply build.sbt]: Done

File: build.sbt [hunk 3]
```diff
@@ -282,4 +284,125 @@ val generate = taskKey[Unit]("generate code from APIs")
...
+    }
+  },
+  licenses += ("Apache-2.0", url("http://www.apache.org/licenses/LICENSE-2.0")),
+  pomExtra :=
+    <url>https://www.unitycatalog.io/</url>
+      <scm>
+        <url>git@github.com:unitycatalog/unitycatalog.git</url>
+        <connection>scm:git:git@github.com:unitycatalog/unitycatalog.git</connection>
+      </scm>
+      <developers>
```
[the remark this answers is not in the document]
Comment (ravivj-db) [reply build.sbt]: Removed.

File: build.sbt [hunk 4]
```diff
@@ -125,6 +128,7 @@ lazy val client = (project in file("clients/java"))
       val _ = openApiGenerate.value
     }
   )
+  .settings(javaOnlyReleaseSettings: _*)
```
Comment (tdas) [inline build.sbt:131]: you dont need this. you can just put the javaOnlyReleaseSettings without expansion in the previous settings block. see https://github.com/delta-io/delta/blob/master/build.sbt#L589

Comment (ravivj-db) [reply build.sbt]: Done

**Review by tdas — APPROVED** [review]

**Pull request merged**